spec/unit/policyfile_demands_spec.rb

```diff
@@ -821,24 +821,83 @@
           # This also conflicts, but only via the graphs
           "remote-cb-two" => {
             "1.2.3" => [ ]
+          },
+
+          # This has a dependency on a conflicting cookbook
+          "dep_on_conflicting_cookbook" => {
+
+            # Only the older cookbook has the conflicting dep, however we don't
+            # know how the dependency solver will solve this without doing more
+            # inspection of the graph, so the expected behavior is to error if
+            # any possible solution could have a conflict.
+            "1.0.0" => [ ["remote-cb-two", ">= 0.0.0" ] ],
+            "2.0.0" => [ ]
+
           }
 
         }
       end
 
-      context "and no explicit source is given for the conflicting cookbook" do
+      context "and the conflicting cookbook is in the run list" do
 
-        before do
-          # This is on the community site
-          policyfile.dsl.cookbook("remote-cb")
+        let(:run_list) { [ 'repo-cookbook-one', 'remote-cb', 'remote-cb-two' ] }
+
+        context "and no explicit source is given for the conflicting cookbook" do
+
+          before do
+            # This is on the community site
+            policyfile.dsl.cookbook("remote-cb")
+          end
+
+          it "raises an error describing the conflict" do
+            repo_path = File.expand_path("path/to/repo")
+
+            expected_err = <<-ERROR
+Source supermarket(https://supermarket.chef.io) and chef_repo(#{repo_path}) contain conflicting cookbooks:
```

Comment:
That's forthcoming in a follow-up patch. Right now there just isn't a good way to specify in your policyfile if you want a cookbook to come from a specific supermarket because of implementation details (longer version: you can only put something like `cookbook "foo", artifactserver: "https://supermarket.chef.io/apiv1/cookbooks/foo/1.2.3"` (or whatever the URL is) in a Policyfile.rb).

I proposed a couple different designs in this gist: https://gist.github.com/danielsdeleo/eba534af8d1201e52d7e I think "C" was the winner but for implementation reasons I'll have to do both B and C, with C being an alternative way of expressing the same things as in B.

I have "include instructions for resolving conflicts in error messages" tracked in the acceptance criteria for the ticket.